core/src/modules/ops/record.ts

```diff
@@ -747,114 +757,186 @@ export namespace RecordOps {
    * This method today always returns a record by finding it or making a a new one... unless it throws because the source isn't allowed to make new records.
    */
   export async function findOrCreateByUniqueRecordProperties(
-    hash: {
+    hashes: {
       [key: string]: (string | number | boolean | Date)[];
-    },
-    source?: boolean | Source
+    }[],
+    referenceIds: string[],
+    source?: boolean | Source,
+    includeAllProperties = false
   ) {
```

Comment:
This method is drastically updated to do *everything* in batches now.  It never throws, but returns a response payload with per-record errors if needed